src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -3043,6 +3121,16 @@ public static ForkJoinPool commonPool() {
         return common;
     }
 
+    /**
+     * Package-private access to commonPool overriding zero parallelism
+     */
+    static ForkJoinPool asyncCommonPool() {
+        ForkJoinPool cp; int p;
+        if ((p = (cp = common).parallelism) < 2)
+            U.compareAndSetInt(cp, PARALLELISM, p, 2);
+        return cp;
```

Comment:
Is this okay for deployments that run with java.util.concurrent.ForkJoinPool.common.parallelism set?

Reply:
Yes, but we must override only if set to 0. So changed to:
+        if ((p = (cp = common).parallelism) == 0)
+            U.compareAndSetInt(cp, PARALLELISM, 0, 2);

If users set to 1, some async CompletableFuture usages that used to bypass may run more slowly (or more quickly because of less overhead).

Reply:
I don't think this would help anyone? We override in those existing situations in which threads would have been created anyway via the (now removed) bypass in CompletableFuture (also SubmissionPublisher). So anyone who used them with zero parallelism expecting no internal jdk thread creation will already have been surprised.